src/main/scala/smack/messaging/Client.scala

```diff
@@ -387,8 +398,23 @@ class Client extends FSM[State, Context] {
       sender ! Messages.GetChatRoomsResponse(MultiUserChatManager.getInstanceFor(connection).getHostedRooms(chatService.value).map(c ⇒ ChatRoomId.apply(c.getJid).get).toSet)
       stay
 
-    case Event(Messages.GetJoinedRooms, Context(Some(connection), _, _)) ⇒
-      sender ! Messages.GetChatRoomsResponse(MultiUserChatManager.getInstanceFor(connection).getJoinedRooms().map(c ⇒ ChatRoomId.apply(c).get).toSet)
+    case Event(Messages.GetRoomMembers(room), Context(Some(connection), _, _)) ⇒
+      withChatRoom(room, connection) { chat ⇒
+        val response: akka.actor.Status.Status = Try {
+          sender ! Messages.GetRoomMembersResponse((chat.getMembers map (m ⇒ MemberInfo(UserWithDomain(m.getJid), room))).toSet)
```

Comment:
this gets a bit complex and long. split into a few lines, and/or give some variable names to help the reader?